Makefile

```diff
@@ -226,6 +260,8 @@ ifeq (, $(shell which controller-gen))
 	CONTROLLER_GEN_TMP_DIR=$$(mktemp -d) ;\
 	cd $$CONTROLLER_GEN_TMP_DIR ;\
 	go mod init tmp ;\
+	# Reset go flags to avoid error caused by set -mod=vendor
+	export GOFLAGS=""
```

Comment:
I'd put this on the same line as `go get` to avoid the export (and the dependency on `.ONESHELL`):
```bash
GOFLAGS="" go get sigs.k8s.io/controller-tools/cmd/controller-gen@v0.3.0 ;\
```